DEV-INTRO.md

```diff
@@ -1,59 +1,166 @@
+
+- **lib** holds all the R packages for the specific version of PEcAn and R. This folder will be shared amongst all other containers, and will contain the compiled PEcAn code.
+- **pecan** this holds all the data, such as workflows and any downloaded data.
+- **portainer** if you enabled the portainer service this folder is used to hold persistent data for this service
+- **postgres** holds the actual database data. If you want to backup the database, you can stop the postgres container, zip up the folder.
+- **rabbitmq** holds persistent information of the message broker (rabbitmq). 
+- **traefik** holds persisent data for the web proxy, that directs incoming traffic to the correct container.
+
+These folders will hold all the persistent data for each of the respective containers and can grow. For example the postgres database is multiple GB. The pecan folder will hold all data produced by the workflows, including any downloaded data, and can grow to many giga bytes.
+
+#### postgresql database
+
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
+
+Once the database has finished starting up we will initialize the database using: `docker run --rm --network pecan_pecan pecan/db`. Once that is done we create two users for BETY:
 
 ```
-library(devtools)
-install_github("pecan", "PEcAnProject")
+# guest user
+docker-compose run --rm bety user guestuser guestuser "Guest User" guestuser@example.com 4 4
+
+# example user
+docker-compose run --rm bety user carya illinois "Carya Demo User" carya@example.com 1 1
 ```
 
-#### "Makefile"
+#### copy web config file
+
+The `docker-compose.dev.yaml` file has a section that will eanble editing the web application. This is by default commented out. If you want to uncoment it you will need to first copy the config.php from the docker/web folder. You can do this using `cp docker/web/config.docker.php web/config.php`.
+
+#### copy R packages
+
+Next copy the R packages from a container to your local machine as the `volumes/lib` folder. This is not really needed, but will speed up the process of the first compilation. Later we will put our newly compiled code here as well. 
+
+You can copy all the data using `docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -r /usr/local/lib/R/site-library/* /rlib/`. This will copy all compiled packages to your local machine.
```

Comment:
@robkooper I'm getting an error from this step: `cp: cannot stat '/usr/local/lib/R/site-library/*': No such file or directory`. Which is accurate, as I only have a path `/usr/local/lib`. 

I also don't really understand what this step is doing, I think because it's mixing a Docker command with `cp`. Is `docker run` compiling the R packages or something?

Reply:
are you running that command inside docker, or on your local machine? docker run does not compile the files, however the pecan/base:develop image should have all the R files already compiled.

Reply:
I think on my local machine? I'm running the command in my local Terminal from within my local pecan directory.

Reply:
make sure to run 
```
docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -r /usr/local/lib/R/site-library/* /rlib/
```

It copies the data from the container to your local machine.

Reply:
I think that's what I did, but that's what isn't working and I don't understand what it's doing to be able to troubleshoot: 
```
(base) Kristinas-MBP:pecan kristinariemer$ pwd
/Users/kristinariemer/Dropbox/Documents/UofA/Projects/PEcAn/pecan
(base) Kristinas-MBP:pecan kristinariemer$ docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -r /usr/local/lib/R/site-library/* /rlib/
cp: cannot stat '/usr/local/lib/R/site-library/*': No such file or directory
```
Is `/usr/local/lib/R/site-library/` supposed to be a local folder?

Reply:
that command looks correct, and the folder `/usr/local/lib/R/site-library` is in the docker container.
can you do a `docker pull pecan/base:develop` and test one more time, while I debug it on my site as well.

Reply:
Doing that docker pull did do some stuff and returned `Status: Downloaded newer image for pecan/base:develop`. But even after doing that, I'm still getting the same "no such file or directory" error from the `docker run`.

Reply:
ok this should work (fingers crossed). This will take a while
```
docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -a /usr/local/lib/R/site-library/. /rlib/
```

Reply:
updated documentation as well.

Reply:
This worked!